framework/doc/content/documentation/application_development/performance_benchmarking.md

```diff
@@ -1,39 +1,93 @@
--------------------
+Performance Benchmarking
+=========================
 
-This script provides functionality for benchmarking and tracking MOOSE performance.  It can be
-used to run benchmarks, generate trend visualizations, and look at stats comparing bencharks
-between various revisions.  Normally, benchmarks should be run through the test harness (i.e.
-using the ``run_tests`` script) by doing e.g. ``./run_tests --run speedtests``
+Utilities for doing performance benchmarking of MOOSE-based applications are included in the main
+MOOSE repository.  These utilities provide functionality for benchmarking and tracking MOOSE
+performance.  They can be used to run benchmarks, generate trend visualizations, and look at stats
+comparing bencharks between various revisions.  The following sections describe how to setup a
+benchmark machine and use it to run benchmarks and visualize results.
+
+Tuning a Benchmarking Machine
+------------------------------
 
 In order to obtain accurate results, you need to run the benchmark process(es)
 as close to isolated as possible.  On a linux system, you should e.g. use cpu
 isolation via setting kernel boot parameters:
 
- ```
+```
 isolcpus=[n] rcu_nocbs=[n]
 ```
 
-in your boot loader (e.g. grub).  The benchmarking tools should automatically
-detect CPU isolation on Linux and schedule benchmark jobs to those CPUs.You
-should also disable any turbo functionality.  For example on ``intel_pstate``
-driver cpus:
+in your boot loader (e.g. grub).  The benchmarking tools/scripts in MOOSE should automatically
+detect CPU isolation on Linux and schedule benchmark jobs to those CPUs. You should also disable
+any turbo functionality.  For example on `intel_pstate` driver cpus:
 
 ```
 $ echo "1" > /sys/devices/system/cpu/intel_pstate/no_turbo
 ```
 
-This will need to be done on every boot.  You can use the sysfsutils package
-and its ``/etc/sysfs.conf`` configuration file to do this.
+You will also want to turn off any hyperthreading for cores you use for benchmarking.  You can do
+this in the bios or by something like:
+
+```
+$ echo "0" > /sys/devices/system/cpu/cpu[n]/online
+```
+
+for each hyperthread core you want running - you can look in `/proc/cpuinfo` for pairs of cpus
+that have the same core id turning off one of the pair.  These will need to be done on every boot.
+You can use the sysfsutils package and its `/etc/sysfs.conf` configuration file to do this
+persistently on boot - i.e.:
+
+```
+devices/system/cpu/intel_pstate/no_turbo = 1
+devices/system/cpu/cpu3/online = 0
+devices/system/cpu/cpu5/online = 0
+```
+
+Test Harness Benchmarks
+------------------------
+
+Benchmarks can be run through the test harness (i.e.  using the `run_tests` script) by doing
+e.g. `./run_tests --run speedtests`.  When this is done, the test harness looks for test spec
+files named `speedtests` just like the `tests` files that contain regular moose test details.
+The format for these files is:
+
+```
+[Benchmarks]
```

Comment:
Can you just include a benchmark file directly?

!listing full/path/from/root/of/repo/benchmark.i

Reply:
I'd rather have a template file that gives an overview of available options.